src/computerAlgorithms.js

```diff
@@ -0,0 +1,202 @@
+      isMarkOn = true;
+      markPosition = i;
+      break;
+    }
+    i += 2;
+  }
+
+  return [isMarkOn, markPosition];
+}
+
+function findRepetitionsAlgorithm(squares, playerGoesFirst){
+  const computerSymbol = playerGoesFirst ? 'O' : 'X';
+  const playerSymbol = computerSymbol === 'X' ? 'O' : 'X';
+
+  if (findRepetitions(squares, playerSymbol) !== null) {
+    return findRepetitions(squares, playerSymbol);
+  } else if (findRepetitions(squares, computerSymbol) !== null) {
+    return findRepetitions(squares, computerSymbol);
+  }
+
+  return null;
+}
+
+function findRepetitions(squares, symbol){
+  for (let i = 0; i < lines.length; i++) {
+    const [a, b, c] = lines[i];
+
+    if (squares[a] === symbol || squares[b] === symbol || squares[c] === symbol) continue;
+
+    if (squares[a] && squares[a] === squares[b]) {
+      return c;
+    } else if (squares[b] && squares[b] === squares[c]) {
+      return a;
+    } else if (squares[a] && squares[a] === squares[c]) {
+      return b;
+    }
+  }
+
+  return null;
+}
+
+function calculateDraw(squares) {
+  for (let i = 0; i < squares.length; i++) {
+    if (!squares[i]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function calculateWinner(squares) {
+  for (let i = 0; i < lines.length; i++) {
+    const [a, b, c] = lines[i];
+    if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
+      const winnerLine = lines[i];
+      const winnerMark = squares[a];
+      return [winnerMark, winnerLine];
+    }
+  }
+  return null;
+}
+
+function getRandomInt(min, max){
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function getRandomAnglePosition(squares){
+  let squareNumber = getRandomInt(0,8);
+  while((squareNumber % 2 !== 0) || (squareNumber === 4) || squares[squareNumber]) {
+    squareNumber = getRandomInt(0,8);
+  }
+  return squareNumber;
+}
+
+function getRandomSidePosition(squares){
+  let squareNumber = getRandomInt(0,8);
+  while((squareNumber % 2 !== 1) || (squareNumber === 4) || squares[squareNumber]) {
+    squareNumber = getRandomInt(0,8);
+  }
+  return squareNumber;
+}
+
+function getPositionNearXOnAngle(squares) {
+  for(let i = 0; i < lines.length; i++) {
+    let [a, b, c] = lines[i];
+
+    if (!squares[a] && squares[b] === 'X' && !squares[c]) {
+      if(Math.round(Math.random())){
+        return a;
+      } else {
+        return c;
+      }
+    }
+  }
+}
+
+function getAnglePositionOnFriendlyLine(squares) {
+  for(let i = 0; i < straightLines.length; i++) {
+    const [a, b, c] = straightLines[i];
+
+    if (squares[a] === 'X' && squares[b] !== 'O' && !squares[c]) {
+      return c;
+    } else if (squares[c] === 'X' && squares[b] !== 'O' && !squares[a]) {
+      return a;
+    }
+  }
+}
+
+function getRandomPosition(squares){
+  let squareNumber = getRandomInt(0,8);
+  while(squares[squareNumber]) {
+    squareNumber = getRandomInt(0,8);
```

Comment:
you don't need to write these params every time, use them as default props in your method

Reply:
Thanks, I fixed it
```
function getRandomPosition(squares){
  let squareNumber = getRandomInt();
  while(squares[squareNumber]) {
    squareNumber = getRandomInt();
  }
  return squareNumber;
}
```